### DISCUSSION TITLE
[Question]: Plugins are sometimes ignored.

### DISCUSSION THREAD

**Discussion by shota1998** [danny-avila/LibreChat#2689] [Q&A] [answered]
### What is your question?

I set up plugins.
I ask the exact same question.
Sometimes the plugins are used, and other times they are ignored.

Is this a specification or a bug?
If it's a specification, what are the criteria for being ignored?
Also, is it possible to force the use of plugins?

If it's a bug, is it already recognized?

PS: The path to the code of conduct is returning a 404 error.

### More Details

- endpoint: azure open ai
- plugin: azure ai search

### What is the main subject of your question?

Other

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/41133886/f2fe2d05-5a24-4fad-9f75-30cd0735b596)
![image](https://github.com/danny-avila/LibreChat/assets/41133886/64f24686-f54d-4ba2-9df3-14ef0f36f9f8)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Yes, this is expected behavior. It's all about prompting, especially in the custom instructions. 

It should be possible in the future to "bind" tools when I work on a tooling rework, available to both OpenAI and non-OpenAI endpoints.

However, only OpenAI has this feature that I know of: https://platform.openai.com/docs/api-reference/chat/create#chat-create-tool_choice
